specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/ContainerAppsBuilds.json

```diff
@@ -0,0 +1,403 @@
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ContainerAppsNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "ARM operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/ContainerAppsBuildCollection"
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "ContainerAppsBuilds_ListByContainerApp_0": {
+            "$ref": "./examples/ContainerAppsBuilds_ListByContainerApp.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/containerApps/{containerAppName}/builds/{buildName}": {
+      "get": {
+        "tags": [
+          "ContainerAppsBuilds"
+        ],
+        "operationId": "ContainerAppsBuilds_Get",
+        "description": "Get a ContainerAppsBuild resource",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ContainerAppsNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ContainerAppsBuildNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "ARM operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/ContainerAppsBuildResource"
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ContainerAppsBuilds_Get_0": {
+            "$ref": "./examples/ContainerAppsBuilds_Get.json"
+          }
+        }
+      },
+      "delete": {
```

Comment:
why is `put` not supported? how is this resource created?

Reply:
Sure thing!

For both ContainerAppsBuilds and ContainerAppsPatches, the proxy resources are created by services ourselves. ContainerAppsBuilds are created when user uploads a file to the upload endpoint of container app, ContainerAppsPatches are created when the control plane retrieves new detected vulnerability reports generated in data plane. 

ContainerAppsPatches allow users to interact with it by applying, skipping, or deleting a patch. ContainerAppsBuilds is a representation of the image-building process when users upload their files to the specific container app.

Reply:
Unfortunately, the `/upload` is the only way to trigger the creation of this proxy resource. Reason being the build is more of a service that automatically gets triggered when users upload their source codes.

Similarly, patches are not created by user, but created when provider detected new potential patches the users can apply.

fyi, this type of design was introduced and approved in the past in #25446. We're aiming to add patching back specified in that PR, but move it under `Microsoft.app/containerApps` rather than `Microsoft.App/builders`. Meanwhile, we decided to adopt this design since `ContainerAppsBuilds` gets created and managed in a very similar fashion.